src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/PartialViewResultExecutor.cs

```diff
@@ -69,16 +71,51 @@ public virtual ViewEngineResult FindView(ActionContext actionContext, PartialVie
             var viewEngine = viewResult.ViewEngine ?? ViewEngine;
             var viewName = viewResult.ViewName ?? actionContext.ActionDescriptor.Name;
 
-            var result = viewEngine.FindPartialView(actionContext, viewName);
+            var result = viewEngine.GetView(executingFilePath: null, viewPath: viewName, isMainPage: false);
+            var originalResult = result;
+            if (!result.Success)
+            {
+                result = viewEngine.FindView(actionContext, viewName, isMainPage: false);
+            }
+
+            if (!result.Success)
+            {
+                if (originalResult.SearchedLocations.Any())
+                {
+                    if (result.SearchedLocations.Any())
+                    {
+                        // Return a new ViewEngineResult listing all searched locations.
+                        var locations = new List<string>(originalResult.SearchedLocations);
+                        locations.AddRange(result.SearchedLocations);
+                        result = ViewEngineResult.NotFound(viewName, locations);
+                    }
+                    else
+                    {
+                        // GetView() searched locations but FindView() did not. Use first ViewEngineResult.
```

Comment:
> The important thing is just that one ViewEngineResult has useful information for the caller and the other doesn't.

But the current API doesn't express this. We're repurposing the `SearchedLocations` to determine this. A proper API that expressively does this would be better. Avoiding extra `ViewEngineResult` instances seems tangential.